dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layers/PoolingLayer.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2015 Seoul National University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.snu.dolphin.dnn.layers;
+
+import edu.snu.dolphin.dnn.blas.Matrix;
+import edu.snu.dolphin.dnn.blas.function.Function;
+import edu.snu.dolphin.dnn.conf.LayerConfigurationParameters;
+import edu.snu.dolphin.dnn.layerparam.initializer.LayerParameterInitializer;
+import org.apache.reef.tang.annotations.Parameter;
+
+import javax.inject.Inject;
+
+/**
+ * Pooling layer.
+ *
+ * This layer is not learnable.
+ * This layer resizes input matrix spatially, using max pooling or average pooling.
+ * In a forward pass,
+ * max pooling picks maximum value in certain range ( kernalHeight * kernalWidth) and these values make up output.
+ * average pooling get average of values in certain range (kernalHeight * kernalWidth) and these values make up output.
+ * In a backward pass,
+ * each value of error matrix is the sum of elements in next error matrix
+ * on which had an impact in feedforward step.
+ */
+
+public abstract class PoolingLayer extends LayerBase {
```

Comment:
If you have an idea to implement `AveragePoolingLayer` and `MaxPoolingLayer` derived from this `PoolingLayer`, `PoolingLayer` can be an abstract class, leaving `feedForward()` and `backPropagate()` as abstract functions.
Otherwise, we can throw an exception in the body of `feedForwardMaxPooling()` and `feedForwardAveragePooling()`, for example, `RuntimeException("Not implemented")`. In the later case, we can remove `abstract` keyword from the class declaration and `feedFoward*Pooling()` functions. `feedFoward*Pooling()` functions should be `private` in this case.